contracts/ReentrancyGuard.sol

```diff
@@ -30,10 +21,9 @@ contract ReentrancyGuard {
    * wrapper marked as `nonReentrant`.
    */
   modifier nonReentrant() {
-    require(reentrancyLock == REENTRANCY_GUARD_FREE);
-    reentrancyLock = REENTRANCY_GUARD_LOCKED;
+    uint localCounter = ++guardCounter;
```

Comment:
I think this would read better if we split the increment and the assignment, the pre-increment syntax might be a bit too magic for something as sensitive as this.

```
guardCounter += 1;
uint256 localCounter = guardCounter;
```